### PR TITLE
fix multiple header matches in root vs

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/47148

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/model/virtualservice.go [hunk 1]
```diff
@@ -429,6 +429,35 @@ func mergeHTTPMatchRequest(root, delegate *networking.HTTPMatchRequest) *network
 	return out
 }
 
+func cloneHTTPMatchRequest(delegate *networking.HTTPMatchRequest) *networking.HTTPMatchRequest {
+	copied := &networking.HTTPMatchRequest{}
```
Comment (howardjohn) [inline pilot/pkg/model/virtualservice.go:433]: should we use `proto.Clone`? Might be more reliable?

Comment (ramaraochavali) [reply pilot/pkg/model/virtualservice.go:433]: I started with that first but later changed to this thinking that this performs better.

Comment (hzxuzhonghu) [reply pilot/pkg/model/virtualservice.go:433]: like https://github.com/istio/istio/pull/47214/files#diff-265bcfd360a47f223b24a3151abaed4b59145e8b1923dd3e68e0518bd2e50dbeR384

Here we only need shallow copy

Comment (ramaraochavali) [reply pilot/pkg/model/virtualservice.go:433]: I do not know if it is safe with protobuf with locks etc?

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
we need backport

**Pull request merged**